_posts/2024-03-02-social-integration.md

```diff
@@ -0,0 +1,177 @@
+From our point of view, we don't actually mind either way: as long as our players can download it for free and enjoy the content, we are more than happy and grateful for your efforts.
+
+The plugin system that is added to 14.0 uses the same logic and mechanism.
+Although technically it is very different, more on that later, but from your point of view it is the same: download a plugin, put it in the right folder, and enjoy the added functionality.
+
+Currently we are releasing three plugins: Steam, Discord, and GOG Galaxy.
+The source of those plugins are released under MIT licenses; another Open Source license, with much simpler conditions compared to GPLv2.
+
+## Security is a thing
+
+Content uploaded to BaNaNaS is executed in the game via sandboxes: nobody can read files on your computer, make network connections, or anything like that.
+They can only do things we allow the addons to do, which is a very restrictive set of actions.
+
+This is a lot harder for a plugin system that integrates with platforms like Steam, Discord, GOG Galaxy, etc.
+We can't sandbox those really, as they have to be native binaries (executables) that run on the player's system.
+
+So, we have a dilemma: if we would allow a plugin system in the same way we allow BaNaNaS content, anyone could upload a plugin for anyone to download.
+And that plugin could, in theory, contain things that are harmful to the player.
+For example, a bitcoin miner or something silly.
+
+This meant we had to find a balance between allowing such plugins and not allowing anyone to just provide any plugin.
+What we came up with, is a system where a plugin can only be loaded into OpenTTD, if we, the OpenTTD Development Team, approved the plugin.
+
+We find this less than ideal, as we like that BaNaNaS allows anyone to come up with ideas, try them out, see what people think.
+But, we also have to keep the security of players in mind.
+And there is just too much potential for harm when we would open up the plugin system to anyone.
+As such, we were left with no other choice than to restrict who can make those plugins; or of course, not do it at all (which is even less ideal).
+
+In result, every plugin is released and signed by OpenTTD, and before loaded into the game, it is validated the signature is correct.
+If you are a developer and want to design your own plugin, please come talk to us.
+We are not against new and different plugins; but we have a responsibility to the players to ensure it doesn't compromise their security.
+
+## Capabilities
+
+With the licensing and security finally addressed, and a solid idea how to build the plugin system, we started work.
+But as you can imagine, it was a lot of work, with a lot of moving parts.
+Like ... seriously, it took many hours figuring all this out, and getting it right.
+
+An additional challenge was that we release for three OSes: Windows, Linux and MacOS.
```

Comment:
What do "we" release? Please consider adding the word "_builds_", for example: "An additional challenge was that we release builds for three OSes: Windows, Linux and MacOS."